scripts/validate_rst_title_capitalization.py

```diff
@@ -261,14 +293,20 @@ def main(source_paths: list[str]) -> int:
 
     number_of_errors: int = 0
 
-    for filename in source_paths:
```

Comment:
Instead of this implementation, why you don't simply implement a function that given `source_paths` yields all the files to check? I think it will be much clearer than what you're doing here. The only change you'd need in this function would be to replace this line by `for filename in walk_paths(source_paths):`.

Also, please use `pathlib` instead of `os.path`.

I don't understand why you skip the `png` files. If this is to validate `rst` files, shouldn't we ignore everything that is not `rst`?

Finally, do you know if this script is being called in the CI? Or why is it not? Are all titles correct after your changes? If that's the case, can you start running it, so we make sure all titles are correct at all times.

I don't think the CI errors are related to your changes, if you update your local branch and push, the CI may become green.